aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -254,6 +321,26 @@ module aptos_framework::transaction_validation {
         epilogue_gas_payer(account, addr, storage_fee_refunded, txn_gas_price, txn_max_gas_units, gas_units_remaining);
     }
 
+    /// `epilogue()` then optionally return the amount held in prologue to the gas payer.
+    fun epilogue_gas_deposit(
+        account: signer,
+        storage_fee_refunded: u64,
+        txn_gas_price: u64,
+        txn_max_gas_units: u64,
+        gas_units_remaining: u64,
+        required_deposit: Option<u64>,
+    ) {
+        let gas_payer = signer::address_of(&account);
+        refund_deposit(gas_payer, required_deposit);
```

Comment:
nit: would be great to document this must happen first

Reply:
fixed in func doc

Reply:
fixed in func doc